overlord/devicestate/devicestate_systems_test.go

```diff
@@ -660,3 +660,170 @@ func (s *deviceMgrSystemsSuite) TestRequestAction1618(c *C) {
 	c.Assert(err, ErrorMatches, ".*/seed/systems/20191119: no such file or directory")
 	c.Check(s.logbuf.String(), Equals, "")
 }
+
+func (s *deviceMgrSystemsSuite) TestRebootNoLabelNoModeHappy(c *C) {
+	err := s.mgr.Reboot("", "")
+	c.Assert(err, IsNil)
+
+	m, err := s.bootloader.GetBootVars("snapd_recovery_mode", "snapd_recovery_system")
+	c.Assert(err, IsNil)
+	// requested restart
+	c.Check(s.restartRequests, DeepEquals, []state.RestartType{state.RestartSystemNow})
+	// but no bootloader changes
+	c.Check(m, DeepEquals, map[string]string{
+		"snapd_recovery_system": "",
+		"snapd_recovery_mode":   "",
+	})
+	c.Check(s.logbuf.String(), Matches, `.*: rebooting system\n`)
+}
+
+func (s *deviceMgrSystemsSuite) TestRebootLabelAndModeHappy(c *C) {
+	s.state.Lock()
+	s.state.Set("seeded-systems", []devicestate.SeededSystem{
+		{
+			System:  s.mockedSystemSeeds[0].label,
+			Model:   s.mockedSystemSeeds[0].model.Model(),
+			BrandID: s.mockedSystemSeeds[0].brand.AccountID(),
+		},
+	})
+	s.state.Unlock()
+
+	err := s.mgr.Reboot("20191119", "install")
+	c.Assert(err, IsNil)
+
+	m, err := s.bootloader.GetBootVars("snapd_recovery_mode", "snapd_recovery_system")
+	c.Assert(err, IsNil)
+	c.Check(m, DeepEquals, map[string]string{
+		"snapd_recovery_system": "20191119",
+		"snapd_recovery_mode":   "install",
+	})
+	c.Check(s.restartRequests, DeepEquals, []state.RestartType{state.RestartSystemNow})
+	c.Check(s.logbuf.String(), Matches, `.*: rebooting into system "20191119" in "install" mode\n`)
+}
+
+func (s *deviceMgrSystemsSuite) TestRebootModeOnlyHappy(c *C) {
+	s.state.Lock()
+	s.state.Set("seeded-systems", []devicestate.SeededSystem{
+		{
+			System:  s.mockedSystemSeeds[0].label,
+			Model:   s.mockedSystemSeeds[0].model.Model(),
+			BrandID: s.mockedSystemSeeds[0].brand.AccountID(),
+		},
+	})
+	s.state.Unlock()
+
+	for _, mode := range []string{"recover", "install"} {
+		s.restartRequests = nil
+		s.bootloader.BootVars = make(map[string]string)
```

Comment:
a followup if you prefer, but in general we should be moving away from setting BootVars directly in tests and just use SetBootVars instead, because this will make it easier to swap out different Bootloader impls which while not a big deal for this code becomes tricky easily for other code

Reply:
leaving this alone for now

Reply:
I will do this (and the one below) in a followup